src/xrpld/app/tx/detail/MPTokenIssuanceSet.cpp

```
@@ -147,6 +183,24 @@ MPTokenIssuanceSet::doApply()
     if (flagsIn != flagsOut)
         sle->setFieldU32(sfFlags, flagsOut);
 
+    if (domainID)
+    {
+        // This is enforced in preflight.
+        XRPL_ASSERT(
+            sle->getType() == ltMPTOKEN_ISSUANCE,
+            "MPTokenIssuanceSet::doApply : modifying MPTokenIssuance");
+
+        if (*domainID != beast::zero)
+        {
+            sle->setFieldH256(sfDomainID, *domainID);
+        }
+        else
```

In this case when `domainID` flag is set, but the value of `domainID` is zero, should we return an error instead of `tesSUCCESS`?